File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let begin = 0, res = 0, k = 1
    for(let end = 0; end<nums.length; end++){
        if(nums[end]== 0)k--
        if(k<0){
            if(nums[begin] == 0){
                k++
            }
            begin++
        }
        let len = end - begin
        res = Math.max(len, res)
    }
    return res
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        begin, max_len, k = 0, 0, 1
        for end in range(len(nums)) : 
            if nums[end] != 1 : k-=1
            while k < 0 :
                if nums[begin] == 0 : k+=1
                begin+=1
            length = end - begin
            max_len = max(max_len, length)
        return max_len